cypress-io/cypress
# feat: debug status badge

DRAFT: Need to add spider icon to design system and incorporate

- Closes #24852

### User facing changelog
Adds status badge to Debug Sidebar, indicating the results of the latest run.

### Additional details
Did some refactoring of the `SidebarNavigation.vue` component, lifting the queries up to a container component to make the component easier to test.

### Steps to test
Driven mostly off of the CT tests, if you have `cypress-services` configured locally you can:

1. Check out the `feature/CYCLOUD-665-IATR` and run `yarn`
2. Run `yarn docker-compose && yarn dev`
3. Visit `localhost:3000` and create a project
4. In the Cypress monorepo, checkout this branch and run `CYPRESS_INTERNAL_ENV=development yarn cypress:open`
5. Select a project and add your newly created `projectId`
6. Tweak tests to cause a failure and run `npx cypress run --record --key <RECORD_KEY>`
7. If you've recorded different runs, you can change the run by changing `runByNumber(runNumber: 6) {` to whatever run you want to target (not dynamic yet)

### How has the user experience changed?
<img width="595" alt="Screen Shot 2022-12-14 at 12 14 31 PM" src="https://user-images.githubusercontent.com/25158820/207674678-65bb86db-14f8-4d9c-a3e1-b0be7a8a2dae.png">
<img width="662" alt="Screen Shot 2022-12-14 at 12 15 28 PM" src="https://user-images.githubusercontent.com/25158820/207674681-04466bca-3603-48ad-a835-69b8ad9bc80d.png">
<img width="640" alt="Screen Shot 2022-12-14 at 12 16 22 PM" src="https://user-images.githubusercontent.com/25158820/207674685-8a204280-8bce-44d8-92e5-21199adf5166.png">


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
